src/libraries/System.Net.Quic/src/System/Net/Quic/Interop/msquic.cs

```diff
@@ -87,6 +87,7 @@ public static bool StatusFailed(int status)
         public static int QUIC_ADDRESS_FAMILY_UNSPEC => OperatingSystem.IsWindows() ? MsQuic_Windows.QUIC_ADDRESS_FAMILY_UNSPEC : (OperatingSystem.IsLinux() || OperatingSystem.IsAndroid()) ? MsQuic_Linux.QUIC_ADDRESS_FAMILY_UNSPEC : MsQuic_Linux.QUIC_ADDRESS_FAMILY_UNSPEC;
         public static int QUIC_ADDRESS_FAMILY_INET => OperatingSystem.IsWindows() ? MsQuic_Windows.QUIC_ADDRESS_FAMILY_INET : (OperatingSystem.IsLinux() || OperatingSystem.IsAndroid()) ? MsQuic_Linux.QUIC_ADDRESS_FAMILY_INET : MsQuic_Linux.QUIC_ADDRESS_FAMILY_INET;
         public static int QUIC_ADDRESS_FAMILY_INET6 => OperatingSystem.IsWindows() ? MsQuic_Windows.QUIC_ADDRESS_FAMILY_INET6 : (OperatingSystem.IsLinux() || OperatingSystem.IsAndroid()) ? MsQuic_Linux.QUIC_ADDRESS_FAMILY_INET6 : MsQuic_Linux.QUIC_ADDRESS_FAMILY_INET6;
+        public static int QUIC_STATUS_ADDRESS_NOT_AVAILABLE => OperatingSystem.IsLinux() ? MsQuic_Linux.QUIC_STATUS_ADDRESS_NOT_AVAILABLE : OperatingSystem.IsMacOS() ? MsQuic_MacOS.QUIC_STATUS_ADDRESS_NOT_AVAILABLE : MsQuic_Linux.QUIC_STATUS_ADDRESS_NOT_AVAILABLE;
```

Comment:
Is this coming from msquic? Do they already have this in their codebase?

Reply:
Yes I added it in msquic https://github.com/microsoft/msquic/pull/3031

Reply:
What about Windows? is there a windows equivalent status code which can be returned in that situation?

Reply:
I think the equivalent is WSAEADDRNOTAVAIL 10049

Reply:
what do you think @rzikm about the WSAEADDRNOTAVAIL 10049 ?

Reply:
From what I can see online this looks like a reasonable choice, it would be nice to test it though, e.g. on a system with Ipv6 disabled and trying to connect to IPv6 address.

Reply:
I tried to reproduce the error locally on my windows machine : I got this error 0x80072743 WSAENETUNREACH.

</html>
</body>
<!--StartFragment-->
<body>

<html>

WSAENETUNREACH10051 | Network is unreachable.A socket operation was attempted to an unreachable network. This usually means the local software knows no route to reach the remote host.
-- | --


<!--EndFragment-->

The problem is that this error seems a little too broad, I don't know if it's a good equivalent for EADDRNOTAVAIL on POSIX

Reply:
@rzikm should I go with WSAENETUNREACH as equivalent of EADDRNOTAVAIL ?

Reply:
I am not sure :/ @nibanks, what do you think? is there EADDRNOTAVAIL equivalent on Windows?

Reply:
I'm not sure. The best way to be sure is an automated test.

Reply:
I'm trying to create an automated test, but I have few questions:
* Does the build machines have Ipv6 addresses, without it I can't request another ipv6 address? connections on ::1 always succeeds on windows even if ipv6 is disabled so I can't use it for the test
* Can I use PowerShell inside my test with Admin privileges to disable ipv6 inside the test and enable it after its end ? didn't find a .NET api that serves this purpose

Reply:
I tried doing another local test by using winsock C api, I got the error WSAEHOSTUNREACH which is kind of like WSAENETUNREACH https://docs.microsoft.com/en-us/windows/win32/winsock/windows-sockets-error-codes-2

Reply:
> I'm trying to create an automated test, but I have few questions:
> 
> * Do the build machines have Ipv6 addresses, without it I can't request another ipv6 address (if I read the docs carefully) ? connections on ::1 always succeeds on windows even if ipv6 is disabled so I can't use it for the test
> * Can I use PowerShell inside my test with Admin privileges to disable ipv6 inside the test and enable it after its end ? didn't find a .NET api that serves this purpose

@rzikm @wfurt any ideas about whether build machines have assigned ipv6 addresses and the possibility of using powershell in my validation test ?

Reply:
I think they may but I would not count on that in general. I think we want tests to pass without IPv6 available. 
If needed we can submit test to should that as well as @MattGal would probably just know.
It is also possible that there is no simple answer. I could imagine that some old Windows versions don't have IPv6 configured while the new one do. 
What site are you trying to connect to? I check and our Azure test services don't resolve to IPv6. We can possibly fix that so we have better IPv6 coverage in general.

Reply:
>Do the build machines have Ipv6 addresses, without it I can't request another ipv6 address (if I read the docs carefully) ? connections on ::1 always succeeds on windows even if ipv6 is disabled so I can't use it for the test

Both Azure Devops hosted and 1ES-Hosted-pool build machines should have IPv6 enabled on them, but there won't be a public address.  This is probably acceptable for your test purposes but I wanted to be as specific as possible.

> Can I use PowerShell inside my test with Admin privileges to disable ipv6 inside the test and enable it after its end ? didn't find a .NET api that serves this purpose

Probably? The build agents are all single-use, and you can run admin-level powershell commands to modify the machine as much as you like without changing anyone else's build.  However, if anything you do breaks the connection between the Azure DevOps agent and the server, your build will fail  (and may take a long time to report this)

Reply:
> Probably? The build agents are all single-use, and you can run admin-level powershell commands to modify the machine as much as you like without changing anyone else's build. However, if anything you do breaks the connection between the Azure DevOps agent and the server, your build will fail (and may take a long time to report this)

I was primarily thinking about test that would dump `ipconfig /all` and routing table. 

And I think you answered the question @MattGal. Without public IPv6, it won't be able to reach Internet sites.

Reply:
thanks for your inputs
> What site are you trying to connect to? 

I was thinking of using bing.com ipv6 address for the test.

Reply:
Sorry for the delay, I pushed a test

Reply:
@rzikm the test ReturnWSAENETUNREACH_WhenIpv6Disabled had passed, it looks like the status code for Windows is 10051 WSAENETUNREACH. Should I use it now as equivalent of POSIX EAFNOSUPPORT ?

Reply:
Yes, lets do it.

Next step is then creating API proposal, something like https://github.com/dotnet/runtime/issues/74755

Reply:
The proposal #76435 is created